app/Services/TimelineService.php

```diff
@@ -57,4 +58,31 @@ public function reorderEra(TimelineEra $era, array $ids)
             }
         }
     }
+
+    /**
+     * @param array $data
+     * @return bool
+     */
+    public function reorder(ReorderTimelineEras $request): bool
+    {
+        $ids = $request->get('timeline_era');
+        if (empty($ids)) {
+            return false;
+        }
+
+        $position = 1;
+        foreach ($ids as $id) {
+            /** @var TimelineEra $link */
+            $link = TimelineEra::where('id', $id)->first();
```

Comment:
You can replace `where('id', $id)->first()` with simply `find($id)`. Also renaming `$link` to the correct model will make reading the code easier to understand :)